File: FrontEnd/client-app/src/app/shared/model/Lesson.ts
/**
* Model cho lesson để lưu lên server
 * @export
* @class EventForInsert
*/
export class Lesson {
    // 
    ID: number
    // 
    LessonName: string
    // 
    TopicID?: number
    // lưu dư thừa tên chủ đề
    TopicName: string
    // Lưu dư thừa để lấy bài học theo chương trình học
    ProgramID?: number
    // trạng thái: 1 kích hoạt
    Status?: number
    // mô tả bài học
    Description: string
    // lưu ảnh tiết học
    Image: string
    // Link bài học
    LinkURL: string
}

File: BackEnd/ELearningAPI/ELearningAPI/TMSBO/Model/Lesson.cs
using Library;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TMSBO.Model
{
    [ConfigTable("lesson", "", false, "")]
    public class Lesson : BaseModel
    {
        [Key]
        /// <summary>
        /// 
        /// </summary>
        public int ID { get; set; }
        /// <summary>
        /// 
        /// </summary>
        public string LessonName { get; set; }
        /// <summary>
        /// 
        /// </summary>
        public int? TopicID { get; set; }
        /// <summary>
        /// lưu dư thừa tên chủ đề
        /// </summary>
        public string TopicName { get; set; }
        /// <summary>
        /// Lưu dư thừa để lấy bài học theo chương trình học
        /// </summary>
        public int? ProgramID { get; set; }
        /// <summary>
        /// trạng thái: 1 kích hoạt
        /// </summary>
        public int? Status { get; set; }
        /// <summary>
        /// mô tả bài học
        /// </summary>
        public string Description { get; set; }
        /// <summary>
        /// lưu ảnh tiết học
        /// </summary>
        public string Image { get; set; }
        /// <summary>
        /// Link bài học
        /// </summary>
        public string LinkURL { get; set; }
    }
}

